### ISSUE TITLE
Signup: Screen reader issues with About step

### ISSUE THREAD

**Issue by ryelle** [Automattic/wp-calypso#21654]
I was testing signup with VoiceOver and Safari, and ran into some trouble on this screen, `/start/business/about`. The form inputs are not clearly labeled for a screen reader.

> What would you like to name your site?

This one worked great 🙂 

> What will your site be about?

This label is not associated with the input, so the question is never read. This can be fixed by adding  `htmlFor=siteTopic` to the label. I also had no idea until creating this issue that there was an autocomplete.

> What’s the primary goal you have for your site?

This was also not read aloud. I got the idea of it because each checkbox item was read, but it wasn't clear why I could check multiple items, or what this did. I think this can be fixed by using `FormLegend` instead of `FormLabel`. There should be a 1:1 match between input & label, and related inputs can be grouped in fieldsets (we're already doing this), and the description of a fieldset is the legend. See an example on [WebAIM's info on accessible checklists](https://webaim.org/techniques/forms/controls#checkbox)

> How comfortable are you with creating a website?

Again, not read aloud. Same suggestion as above about using `FormLegend`, but also expanding on what 1, 2, 3… are for screen readers – this read to me as "radio button 1 of 5, 1; radio button 2 of 5, 2" etc. We could maybe use [`ScreenReaderText`](https://wpcalypso.wordpress.com/devdocs/design/screen-reader-text) to give more context in the text that's read aloud.

**Comment by fditrapani** [comment]
Thanks for sharing this @ryelle. I'll take care of it when I get some time.

**Comment by gwwar** [comment]
@fditrapani are you still actively working on this one?

**Comment by fditrapani** [comment]
@gwwar No, I'm going to unassign myself as I haven't had a chance to get to it.
